src/qt/trafficgraphwidget.cpp

```diff
@@ -135,32 +161,35 @@ void TrafficGraphWidget::updateRates()
     nLastBytesIn = bytesIn;
     nLastBytesOut = bytesOut;
 
-    while(vSamplesIn.size() > DESIRED_SAMPLES) {
+    while(vSamplesIn.size()  > FLT_MAX - 1) {
```

Comment:
how much memory does this need?

Reply:
I agree - I have been thinking about how to do this. Maybe pruning the list at larger intervals - keeping samples that are above a moving average. Open to suggestions on implementation. :) 

Thanks for looking at this!

Reply:
We can chunk the list at DESIRED_SAMPLES intervals - calculate the moving average - pruning samples that fall below it.